File: SDKs/amazonaws.com/appmesh/2019-01-25/java/lib/src/main/java/openapisdk/models/shared/GatewayRouteData.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * GatewayRouteData
 * An object that represents a gateway route returned by a describe operation.
**/
public class GatewayRouteData {
    @JsonProperty("gatewayRouteName")
    public String gatewayRouteName;
    public GatewayRouteData withGatewayRouteName(String gatewayRouteName) {
        this.gatewayRouteName = gatewayRouteName;
        return this;
    }
    @JsonProperty("meshName")
    public String meshName;
    public GatewayRouteData withMeshName(String meshName) {
        this.meshName = meshName;
        return this;
    }
    @JsonProperty("metadata")
    public ResourceMetadata metadata;
    public GatewayRouteData withMetadata(ResourceMetadata metadata) {
        this.metadata = metadata;
        return this;
    }
    @JsonProperty("spec")
    public GatewayRouteSpec spec;
    public GatewayRouteData withSpec(GatewayRouteSpec spec) {
        this.spec = spec;
        return this;
    }
    @JsonProperty("status")
    public GatewayRouteStatus status;
    public GatewayRouteData withStatus(GatewayRouteStatus status) {
        this.status = status;
        return this;
    }
    @JsonProperty("virtualGatewayName")
    public String virtualGatewayName;
    public GatewayRouteData withVirtualGatewayName(String virtualGatewayName) {
        this.virtualGatewayName = virtualGatewayName;
        return this;
    }
}

File: SDKs/amazonaws.com/appmesh/2019-01-25/python/src/sdk/models/shared/gatewayroutedata.py
import dataclasses
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import resourcemetadata as shared_resourcemetadata
from ..shared import gatewayroutespec as shared_gatewayroutespec
from ..shared import gatewayroutestatus as shared_gatewayroutestatus


@dataclass_json
@dataclasses.dataclass
class GatewayRouteData:
    r"""GatewayRouteData
    An object that represents a gateway route returned by a describe operation.
    """
    
    gateway_route_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('gatewayRouteName') }})
    mesh_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('meshName') }})
    metadata: shared_resourcemetadata.ResourceMetadata = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('metadata') }})
    spec: shared_gatewayroutespec.GatewayRouteSpec = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('spec') }})
    status: shared_gatewayroutestatus.GatewayRouteStatus = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    virtual_gateway_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('virtualGatewayName') }})
    
